Add glossary for typically misused terms (#497)
open-telemetry/opentelemetry-specification: specification/glossary.md

# Glossary

Below are a list of some OpenTelemetry specific terms that are used across this
specification.

## (Telemetry) Library

Denotes the library that implements the *OpenTelemetry API*.

> See [Library resource semantic conventions](data-resource-semantic-conventions.md#library)

Reviewer: from https://github.com/open-telemetry/opentelemetry-specification/pull/494#discussion_r387142820:
```suggestion
## Telemetry SDK

Denotes the library that implements the *OpenTelemetry API*.

See [Library Guidelines](library-guidelines.md#sdk-implementation) and
[Library resource semantic conventions](data-resource-semantic-conventions.md#telemetry-sdk)
```

# Glossary

Below are a list of some OpenTelemetry specific terms that are used across this
specification.

## (Telemetry) Library

> Denotes the actual *OpenTelemetry* (or API-compatible) library in use.

Reviewer 2: ```suggestion
Denotes the library that implements the *OpenTelemetry API*.
```

See [Library resource semantic conventions](data-resource-semantic-conventions.md#library)

<a name="instrumented_library"></a>

## Instrumented Library

> Denotes the library that is monitored, i.e. the library of interest that signals (traces, metrics, logs) are emitted for.

Reviewer 2: ```suggestion
Denotes the library for which the telemetry signals (traces, metrics, logs) are gathered.
```

Reviewer 3: I liked the old version better

Reviewer 4: I think both versions are fine, so I'd leave it to to the author to decide this one ;)

Reviewer 2: The old version:
  * poor English (ends sentence with a preposition)
  * uses inaccurate word "emitted". Telemetry may be gathered but never emitted.

Author: Fair enough @yurishkuro; I already applied your suggestion.

The actual OpenTelemetry API calls can happen inside this library but might as well

> happen in a different one (the [Instrumenting Library](#instrumenting_library)).

Reviewer 2: ```suggestion
The calls to the OpenTelemetry API can be done either by the Instrumented Library itself, 
or by another [Instrumenting Library](#instrumenting_library).
```

Example: `org.mongodb.client`.

<a name="instrumenting_library"></a>

## Instrumenting Library

Denotes the library which provides the instrumentation *for* a given library

> (i.e. *for* the [Instrumented Library](#instrumented_library)). Hence,

Reviewer 2: ```suggestion
Denotes the library that provides the instrumentation for a given [Instrumented Library](#instrumented_library). 
```

*Instrumented Library* and *Instrumenting Library* can be identical if a library

> has built-in OpenTelemetry instrumentation, but they don't necessarily have to.

Reviewer 2: ```suggestion
*Instrumented Library* and *Instrumenting Library* may be the same library
if it has built-in OpenTelemetry instrumentation.
```
